src/components/dialog/dialog.atoms.tsx

```diff
@@ -0,0 +1,36 @@
+import { forwardRef } from 'react'
+
+import { DialogBodyProps, DialogFooterProps, DialogContainerProps, DialogTitleProps } from './types'
+import { ElDialogBody, ElDialogContainer, ElDialogFooter, ElDialogTitle } from './style'
+
+const Container = forwardRef<HTMLDialogElement, DialogContainerProps>(({ children, ...rest }, ref) => {
+  return (
+    <ElDialogContainer {...rest} ref={ref}>
+      {children}
+    </ElDialogContainer>
+  )
+})
+
+const Title = ({ title, ...rest }: DialogTitleProps) => (
+  <ElDialogTitle {...rest} title={title}>
+    {title}
+  </ElDialogTitle>
+)
+
+const Body = ({ children, ...rest }: DialogBodyProps) => <ElDialogBody {...rest}>{children}</ElDialogBody>
+
+const Footer = ({ children, ...rest }: DialogFooterProps) => <ElDialogFooter {...rest}>{children}</ElDialogFooter>
+
+// NOTE: We extend the `Container` component with the `Title`, `Body`, and `Footer` components
+// to provide a more convenient way to render the dialog content.
```

Comment:
> why did you do this instead of simply extending the Container atom (and therefore being able to use Dialog instead of Container)?

it's to achieve the pattern that you suggest [in here](https://github.com/reapit/elements/pull/170#discussion_r1792903937), the exported part is only `Dialog` from the "atoms"